enterprise/backend/src/metabase_enterprise/semantic_search/index.clj

```diff
@@ -113,14 +113,18 @@
   [documents]
   (when (seq documents)
     (let [texts (mapv :searchable_text documents)
-          embeddings (embedding/get-embeddings-batch texts)]
-      (batch-update!
-       (fn [db-records]
-         (-> (sql.helpers/insert-into *index-table-name*)
-             (sql.helpers/values db-records)
-             sql/format))
-       documents
-       embeddings))))
+          text->doc (zipmap texts documents)]
```

Comment:
unrelated but I wonder we should avoid forcing evaluation of the full documents reducible. I know we already do this in the entry points with `(into [] documents-reducible)`, but if we're eventually going to need to deal with a reducible with ~500k documents, maybe we need to keep it reducible and only evaluate one batch-at-a-time.

Reply:
absolutely